src/main/java/com/mapzen/jpostal/Config.java

```diff
@@ -0,0 +1,75 @@
+package com.mapzen.jpostal;
+
+import java.util.Objects;
+
+public final class Config {
+    private final String dataDir;
+    private final String libraryFile;
+
+    private Config(final String dataDir, final String libraryFile) {
+        this.dataDir = dataDir;
+        this.libraryFile = libraryFile;
+    }
+
+    public String getDataDir() {
+        return dataDir;
+    }
+
+    public String getLibraryFile() {
+        return libraryFile;
+    }
+
+    void loadLibrary() {
+        if (this.libraryFile != null) {
+            System.load(this.libraryFile);
+        } else {
+            System.loadLibrary("jpostal");
+        }
+    }
+
+    static IllegalArgumentException mismatchException(final Config current, final Config requested) {
+        return new IllegalArgumentException(String.format("Config mismatch: initialized instance uses [%s], but requested [%s]", current, requested));
+    }
+
+    @Override
+    public String toString() {
+        return "Config{" + "dataDir=" + dataDir + ",libraryFile=" + libraryFile + '}';
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        } else if (!(o instanceof Config other)) {
+            return false;
```

Comment:
Minor callout: This `instanceof` pattern syntax sugar was added in Java 14, so continuing to use it would effectively set the minimum JDK version going forward at 14.

This seems fine to me since older JDKs including 8 and 11 are no longer LTS, IIRC.

If we want to drop this and use the older and more broadly-available syntax, that's a quick fix on my side and I have no problem making it!